src/python/platforms/android/flash.py

```diff
@@ -52,6 +56,31 @@
 FLASH_REBOOT_WAIT = 5 * 60
 
 
+def download_latest_build(build_info, image_regexes):
+  """Download the latest build artifact for the given branch and target."""
+  # Check if our local build matches the latest build. If not, we will
+  # download it.
+  build_id = build_info['bid']
+  target = build_info['target']
+  image_directory = environment.get_value('IMAGES_DIR')
+  last_build_info = persistent_cache.get_value(constants.LAST_FLASH_BUILD_KEY)
+  if not last_build_info or last_build_info['bid'] != build_id:
```

Comment:
nit: invert this condition and early return to avoid the extra indent.

i.e. `if last_build_info and last_build_info['bid'] == build_id:`